File: c-sharp-core/fizzbuzz/FizzBuzz.cs
using System.Collections.Immutable;

namespace fizzbuzz;

public static class FizzBuzz
{
    public record Item(int Index, string Value);

    public static IEnumerable<Item> Generate(int min, int max)
    {
        var range = Enumerable.Range(min, max);
        foreach (var index in range)
        {
            var value = "";
            if (index % 3 == 0)
            {
                value += "Fizz";
            }
            if (index % 5 == 0)
            {
                value += "Buzz";
            }
            yield return new Item(Index: index, Value: value);
        }
    }
}

File: Java/src/main/java/it/kluth/fizzbuzz/FizzBuzz.java
package it.kluth.fizzbuzz;

import java.util.HashMap;
import java.util.Map;
import java.util.stream.IntStream;

public class FizzBuzz {

    private FizzBuzz() {
        throw new IllegalStateException();
    }

    public static Map<Integer, String> generate(int min, int max) {
        var collector = new HashMap<Integer, String>();
        for (var index : IntStream.range(min, max + 1).toArray()) {
            var value = "";
            if (index % 3 == 0) {
                value += "Fizz";
            }
            if (index % 5 == 0) {
                value += "Buzz";
            }
            collector.put(index, value);
        }
        return collector;
    }

}
